### PR TITLE
Make is-installed script not exit on error

### PR DESCRIPTION
**Pull request by kubadz**
**Description**
This PR removes the logic responsible for immediately returning an error code from the is-installed script when the Status in Installation CR becomes `Error`. By default, the script will wait forever until the installation will be successful, but there is also an option added allowing to set the timeout after which the script will return an error code. The script additionally fetches logs from the installer if the timeout occurs while the `Error` status is set on Installation CR.
 
Changes proposed in this pull request:

- stop immediately returning an error code from the is-installed script when the Status in Installation CR becomes `Error`
- add timeout option to the is-installed script
- set default timeout in dockerfiles/scripts using is-installed.sh as a part of Kyma installation
- update documentation

**Related issue(s)**

Reasons of introducing those changes are described in #1610

### CODE REVIEW

**Comment by sjanota** [comment]
@kubadz Remember to update Prow jobs

File: installation/scripts/is-installed.sh [hunk 1]
```diff
@@ -33,18 +86,29 @@ do
...
-    echo "----------"
-    exit 1
+    echo "kyma installation error, description: ${DESC}"
+    if [ "$TIMEOUT_SET" -eq 0 ]; then
+      echo "to fetch the logs from the installer execute: kubectl logs -n kyma-installer $(kubectl get pods --all-namespaces -l name=kyma-installer --no-headers -o jsonpath='{.items[*].metadata.name}')"
+    fi
+    if [ "$TIMEOUT_SET" -ne 0 ] && [ "$ITERATIONS_LEFT" -le 0 ]; then
+      echo "timeout reached on kyma installation error. Fetching logs from the installer:"
+      echo "----------"
+      echo "$(kubectl logs -n kyma-installer $(kubectl get pods --all-namespaces -l name=kyma-installer --no-headers -o jsonpath='{.items[*].metadata.name}'))"
```
Comment (Tomasz-Smelcerz-SAP) [inline installation/scripts/is-installed.sh]: Is there a particular reason to use command substitution / subshell here? It's more explicit to just run the command directly.

File: installation/scripts/is-installed.sh [hunk 2]
```diff
@@ -33,18 +86,29 @@ do
...
-      echo "----------"
-      echo "$(kubectl get po --all-namespaces)"
+    if [ "$TIMEOUT_SET" -ne 0 ] && [ "$ITERATIONS_LEFT" -le 0 ]; then
+      echo "timeout reached. Exiting"
+      exit 1
     fi
-    sleep $DELAY
   fi
+  if [ "${VERBOSE}" -eq 1 ]; then
+    echo "$(kubectl get installation/kyma-installation -o yaml)"
```
Comment (Tomasz-Smelcerz-SAP) [inline installation/scripts/is-installed.sh]: Is there a particular reason to use command substitution / subshell here? It's more explicit to just run the command directly.

File: installation/scripts/is-installed.sh (hunk 1 above)
Comment (kubadz) [reply installation/scripts/is-installed.sh]: Probably not. I just left it like it was before. I agree that it should be run directly.

File: installation/scripts/is-installed.sh (hunk 2 above)
Comment (kubadz) [reply installation/scripts/is-installed.sh]: The same answer as in your previous comment.

File: docs/kyma/docs/033-inst-local-installation-scripts.md [hunk 3]
```diff
@@ -78,3 +78,7 @@ The `installation/scripts/installer.sh` script creates the default RBAC role, in
 The script applies the Installation custom resource and marks it with the `action=install` label, which triggers the Kyma installation.
 
 >**NOTE:** The Kyma installation runs in the background. Execute the `./installation/scripts/is-installed.sh` script to follow the installation process.
+
+## The is-installed.sh script
+
+The `installation/scripts/is-installed.sh` script follows the process of Kyma installation. It checks the status on Installation CR and returns on `Installed` status. If there was a timeout set it fetches the logs from installer when the time reaches the timeout. If timeout is not set it waits infinitely.
```
Comment (tomekpapiernik) [inline docs/kyma/docs/033-inst-local-installation-scripts.md]:
```suggestion
The `installation/scripts/is-installed.sh` script shows the status of Kyma installation in real time. The script checks the status of the Installation custom resource. When it detects that the status changed to `Installed`, the script exits. If you define a timeout period and the status doesn't change to `Installed` within that period, the script fetches the installer logs when it reaches the timeout. If you don't set a timeout period, the script waits for the change of the status until you terminate it. 
```

**Pull request merged**